src/pages/index.js

```diff
@@ -87,25 +119,92 @@ const addFormValidator = new FormValidator(
   cardForm
 );
 
+const avatarFormValidator = new FormValidator(
+  constants.validationSettings,
+  avatarForm
+);
+
 const profileInfo = new UserInfo({
   name: ".profile__name",
   description: ".profile__description",
 });
 
-const editProfilePopup = new PopupWithForm("#edit-popup", handleSaveProfile);
-const newCardPopup = new PopupWithForm("#new-card-popup", handleAddCard);
+const editProfilePopup = new PopupWithForm("#edit-popup", handleUpdateProfile);
+
+const updateAvatarPopup = new PopupWithForm(
+  "#avatar-popup",
+  handleUpdateAvatar
+);
+
 const imagePopup = new PopupWithImage("#preview-image-popup");
 
 editFormValidator.enableValidation();
 addFormValidator.enableValidation();
+avatarFormValidator.enableValidation();
 
 // ! ||--------------------------------------------------------------------------------||
 // ! ||                                Gallery Creation                                ||
 // ! ||--------------------------------------------------------------------------------||
 
-const cardGallery = new Section(
-  { items: constants.initialCards, renderer: renderCard },
-  ".gallery__cards"
-);
-
-cardGallery.renderItems();
+api.getInitialCards().then((cards) => {
+  const cardGallery = new Section(
```

Comment:
Needs correction: you can declare the card gallery in the global scope and reassign it here, so you will be able to move the code from lines 155-207 to the global scope

```js
let cardGallery;

api.getInitialCards().then((cards) => {
  cardGallery.  = new Section(..);
  cardGallery.renderItems();
}).catch(..)
```